### PR TITLE
dev-qt/qtwebengine: fix building on x86

### PR DESCRIPTION
**Pull request by stha09**
Closes: https://bugs.gentoo.org/695446
Package-Manager: Portage-2.3.76, Repoman-2.3.16
Signed-off-by: Stephan Hartmann <stha09@googlemail.com>

### CODE REVIEW

**Comment by Pesa** [comment]
I haven't really been following this bug report... have you tried adding `-fPIC`?

**Comment by stha09** [comment]
> I haven't really been following this bug report... have you tried adding `-fPIC`?

No. The problem here is that the linker prints some warnings but `-Wl,--fatal-warnings` is passed, which turns the warnings into errors.

**Comment by stha09** [comment]
> I haven't really been following this bug report... have you tried adding `-fPIC`?

`-fPIC` is already set by build system.

**Comment by Pesa** [comment]
> No. The problem here is that the linker prints some warnings but -Wl,--fatal-warnings is passed, which turns the warnings into errors.

I understand. I suggested `-fPIC` because that DT_TEXTREL warning usually happens when trying to link non-PIC code into a shared library. But as I said, I haven't really investigated the problem.

File: dev-qt/qtwebengine/qtwebengine-5.14.1.ebuild [hunk 1]
```diff
@@ -78,6 +78,8 @@ DEPEND="${RDEPEND}
 	pax_kernel? ( sys-apps/elfix )
 "
 
+PATCHES=( "${FILESDIR}/${P}-disable-fatal-warnings.patch" )
```
Comment (Pesa) [inline dev-qt/qtwebengine/qtwebengine-5.14.1.ebuild]: can you put the bug number in a comment here please?

**Comment by stha09** [comment]
> > No. The problem here is that the linker prints some warnings but -Wl,--fatal-warnings is passed, which turns the warnings into errors.
> 
> I understand. I suggested `-fPIC` because that DT_TEXTREL warning usually happens when trying to link non-PIC code into a shared library. But as I said, I haven't really investigated the problem.

I took a look and I think it comes from assembler code in openh264 (dct.asm).

File: dev-qt/qtwebengine/qtwebengine-5.14.1.ebuild (hunk 1 above)
Comment (stha09) [reply dev-qt/qtwebengine/qtwebengine-5.14.1.ebuild]: sure, done

**Pull request closed without merging**